src/payment/payments.ts

```diff
@@ -80,11 +76,15 @@ export class Payments extends EventTarget {
               provider: invoice.provider,
             }),
           );
-          this.logger?.debug(`New Invoice received for agreement ${invoice.agreementId}. Amount: ${invoice.amount}`);
+          this.logger.debug(`New Invoice received`, {
+            id: invoice.id,
+            agreementId: invoice.agreementId,
+            amount: invoice.amount,
+          });
         }
       } catch (error) {
-        const reason = error.response?.data?.message || error;
-        this.logger?.debug(`Unable to get invoices. ${reason}`);
+        const reason = error.response?.data?.message || error.message || error;
+        this.logger.error(`Unable to get invoices.`, { reason });
```

Comment:
I'm not sure if that shouldn't be a warning. I'll see the UX when testing it

Reply:
This can stay, as per: https://github.com/golemfactory/golem-js/pull/755#discussion_r1452059163